lib/posix/semaphore.c

```diff
@@ -5,8 +5,11 @@
  */
 
 #include <errno.h>
+#include <signal.h>
```

Comment:
I'm a bit concerned about this inconsistence - in Zephyr, this should technically be `<zephyr/posix/signal.h>` but iirc that breaks a couple of tests (if not here, then in another place where `<signal.h>` is used. IIRC, `timer.c`.